### PR TITLE
Fix nested generic constructor

### PR DESCRIPTION
**Pull request by josesimoes**
## Description
- Now properly resolving nested types in CEE_NEWOBJ.
- Update CLAUDE with reasoning and new findings.

## Motivation and Context
- Found another edge case in generics type resolution when adding new unit tests in mscorlib.
- Related with nanoframework/Home#782.

## How Has This Been Tested?
- [tested against nanoframework/CoreLibrary#275].

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [x] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dev Containers (changes related with Dev Containers, has no impact on code or features)
- [ ] Dependencies/declarations (update dependencies or assembly declarations and changes associated, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist

- [x] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).

### CODE REVIEW

**Comment by josesimoes** [comment]
@coderabbitai review

File: src/CLR/Core/Interpreter.cpp [hunk 1]
```diff
@@ -3067,6 +3067,130 @@ HRESULT CLR_RT_Thread::Execute_IL(CLR_RT_StackFrame &stackArg)
...
+                                    }
+                                    else if (NANOCLR_INDEX_IS_VALID(argElem.Class))
+                                    {
+                                        resolvedArgs[a] = argElem.Class;
+                                    }
+                                    else
+                                    {
+                                        allResolved = false;
+                                    }
+                                }
```
[the remark this answers is not in the document]
Comment (josesimoes) [reply src/CLR/Core/Interpreter.cpp:3143]: Fixed!
